exchange_calendars/pandas_extensions/holiday.py

```diff
@@ -0,0 +1,151 @@
+# https://github.com/pandas-dev/pandas/blob/master/pandas/tseries/holiday.py
+
+import warnings
+
+from pandas.tseries.holiday import (
+    Holiday as PandasHoliday,
+    AbstractHolidayCalendar as PandasAbstractHolidayCalendar,
+)
+
+from pandas import (
+    DatetimeIndex,
+    Series,
+    Timestamp,
+)
+
+from pandas.errors import PerformanceWarning
+
+
+class Holiday(PandasHoliday):
+    """
+    This extension allows to have both offset and observance while the original
+    Holiday does not. The order of application is offset => observance.
+    """
+
+    def __init__(
+        self,
+        name,
+        year=None,
+        month=None,
+        day=None,
+        offset=None,
+        observance=None,
+        start_date=None,
+        end_date=None,
+        days_of_week=None,
+    ):
+        super().__init__(
+            name,
+            year,
+            month,
+            day,
+            None,
+            None,
+            start_date,
+            end_date,
+            days_of_week,
+        )
+        self.name = name
+        self.year = year
+        self.month = month
+        self.day = day
+        self.offset = offset
+        self.start_date = (
+            Timestamp(start_date) if start_date is not None else start_date
+        )
+        self.end_date = Timestamp(end_date) if end_date is not None else end_date
+        self.observance = observance
+        assert days_of_week is None or type(days_of_week) == tuple
+        self.days_of_week = days_of_week
```

Comment:
This seems the same as https://github.com/pandas-dev/pandas/blob/master/pandas/tseries/holiday.py#L214-L225. 

You should be able to get away with just 

```
self.offset = offset
self.observance = observance
```
after your super call right?

Reply:
You're right. I just copy/pasted the codes at the first time, and then put that super call later.
I think below change would suffice, as you pointed out.

```suggestion
        self.offset = offset
        self.observance = observance
```